api/client/charms/localcharmclient.go

```diff
@@ -149,24 +149,14 @@ func hasHooksFolderOrDispatchFile(name string) (bool, error) {
 		return false, err
 	}
 	defer zipr.Close()
-	count := 0
 	// zip file spec 4.4.17.1 says that separators are always "/" even on Windows.
 	hooksPath := "hooks/"
 	dispatchPath := "dispatch"
 	for _, f := range zipr.File {
-		if strings.Contains(f.Name, hooksPath) {
-			count++
-		}
-		if count > 1 {
-			// 1 is the magic number here.
-			// Charm zip archive is expected to contain several files and folders.
-			// All properly built charms will have a non-empty "hooks" folders OR
-			// a dispatch file.
-			// File names in the archive will be of the form "hooks/" - for hooks folder; and
-			// "hooks/*" for the actual charm hooks implementations.
-			// For example, install hook may have a file with a name "hooks/install".
-			// Once we know that there are, at least, 2 files that have names that start with "hooks/", we
-			// know for sure that the charm has a non-empty hooks folder.
+		// Plain directories (e.g. 'hooks/') may or may not be listed in ths
+		// zip file. We are looking for something inside hooks/ that is *not* a
+		// directory.
+		if strings.Contains(f.Name, hooksPath) && !f.FileInfo().IsDir() {
```

Comment:
I suspect this would also validate against:

  - `hooks/foo/bar.sh`

Which isn't ideal. This is better than what we had, but can we make it better still?

Reply:
Right, so we want to only match files that are in the top level of the `hooks/` directory?

Also, AFAIK these are the accepted values for a file mode, which would we consider acceptable? For example would a symlink (e.g. to another Bash script) be an acceptable "hook file"?
```
ModeType = ModeDir | ModeSymlink | ModeNamedPipe | ModeSocket | ModeDevice | ModeCharDevice | ModeIrregular
```

Reply:
Yeah, symlinks are allowed.